dotcom-rendering/src/model/enhance-ad-placeholders.ts

```diff
@@ -65,13 +83,16 @@ const insertAdPlaceholders = (elements: FEElement[]): FEElement[] => {
 			currentElement: FEElement,
 			idx: number,
 		): ReducerAccumulator => {
```

Comment:
Not blocking, although the `reduce` works well, it's a bit hard to follow, I think it might be better to refactor to a simple loop with the accumulator in the parent scope?

Reply:
Yeah that might be easier to read! Although IMO that isn't really in the scope of this ticket and I feel like there are enough changes in here already. Maybe we could add this idea to the backlog?

Reply:
> a simple loop with the accumulator in the parent scope

I think this style is discouraged in DCR in favour of a more functional style.
A previous discussion:
https://github.com/guardian/dotcom-rendering/pull/8118/files

Reply:
That's really useful context, thank you for flagging this Ravi!